src/controller/python/chip/clusters/command.cpp

```diff
@@ -181,6 +237,83 @@ PyChipError pychip_CommandSender_SendCommand(void * appContext, DeviceProxy * de
+    std::unique_ptr<CommandSender> sender =
+        std::make_unique<CommandSender>(callback.get(), device->GetExchangeManager(),
+                                        /* is timed request */ timedRequestTimeoutMs != 0, suppressResponse);
+
+    va_list args;
+    va_start(args, n);
+
+    SuccessOrExit(err = sender->SetCommandSenderConfig(config));
+
+    {
+        for (size_t i = 0; i < n; i++)
+        {
+            void * commandPath = va_arg(args, void *);
+            void * tlv         = va_arg(args, void *);
+            int length         = va_arg(args, int);
+
+            python::CommandPath invokeRequestInfoObj;
+            memcpy(&invokeRequestInfoObj, commandPath, sizeof(python::CommandPath));
+            uint8_t * tlvBuffer = reinterpret_cast<uint8_t *>(tlv);
+
+            app::CommandPathParams cmdParams = { invokeRequestInfoObj.endpointId, /* group id */ 0, invokeRequestInfoObj.clusterId,
+                                                 invokeRequestInfoObj.commandId, (app::CommandPathFlags::kEndpointIdValid) };
+
+            CommandSender::AdditionalCommandParameters additionalParams;
+
+            SuccessOrExit(err = sender->PrepareCommand(cmdParams, additionalParams));
+            {
+                auto writer = sender->GetCommandDataIBTLVWriter();
+                VerifyOrExit(writer != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+                TLV::TLVReader reader;
+                reader.Init(tlvBuffer, static_cast<uint32_t>(length));
+                reader.Next();
+                SuccessOrExit(err = writer->CopyContainer(TLV::ContextTag(CommandDataIB::Tag::kFields), reader));
+            }
+
+            SuccessOrExit(err = sender->FinishCommand(timedRequestTimeoutMs != 0 ? Optional<uint16_t>(timedRequestTimeoutMs)
+                                                                                 : Optional<uint16_t>::Missing(),
+                                                      additionalParams));
+
+            // CommandSender provides us with the CommandReference for this associated command. In order to match responses
+            // we have to add CommandRef to index lookup.
+            VerifyOrExit(additionalParams.mCommandRef.HasValue(), err = CHIP_ERROR_UNEXPECTED_EVENT);
+            SuccessOrExit(err = callback->AddCommandRefToIndexLookup(additionalParams.mCommandRef.Value(), i));
+        }
+    }
+
+    SuccessOrExit(err = sender->SendCommandRequest(device->GetSecureSession().Value(),
+                                                   interactionTimeoutMs != 0
+                                                       ? MakeOptional(System::Clock::Milliseconds32(interactionTimeoutMs))
+                                                       : Optional<System::Clock::Timeout>::Missing()));
+
+    sender.release();
+    callback.release();
+
+    if (busyWaitMs)
+    {
+        usleep(busyWaitMs * 1000);
+    }
```

Comment:
This really seems like we should not allow it...

Reply:
Having this is broken and should be replaced with a different primitive/method. Not for you to fix, but problematic that we are copying the bad pattern one more place.

Reply:
Created https://github.com/project-chip/connectedhomeip/issues/30985